src/main/java/com/google/graphgeckos/dashboard/storage/Builder.java

```diff
@@ -16,37 +16,93 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+import org.springframework.data.annotation.Transient;
 
 /**
  * Contains the information retrieved from a single build bot. It is used as a member of BuildInfo...
- * of {@link #BuildInfo BuildInfo}, and for converting ParsedBuildbotData objects
- * to Builder objects, used by BuildInfo.
+ * of {@link BuildInfo}
  */
 @Entity(name = "builder")
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class Builder {
 
+  @Transient
+  private String commitHash;
+
+  @Transient
+  private String branch;
+
+  /* The timestamp of the build. */
+  private String timestamp;
+
+  /* Name of the buildbot. */
   private final String name;
 
-  // The logs of each compilation stage, stored as described by {@link #Log Log}.
-  private final List<Log> logs;
+  /**
+   * The logs of each compilation stage, stored as described by {@link Log}.
+   */
+  private final List<Log> logs = new ArrayList<>();
 
-  // Builder compilation status, as described by {@link #BuilderStatus BuilderStatus}.
-  private final BuilderStatus status;
+  /**
+   * Builder compilation status, as described by {@link BuilderStatus}.
+   */
+  private BuilderStatus status;
```

Comment:
Why isn't the status field final?

Reply:
Because in Java it is not allowed to mark fields as final if they are not assigned in the constructor.

Reply:
The question is, why don't we have the status when this is constructed.